src/hooks/useArticleLinksNavigation.js

```diff
@@ -57,6 +60,15 @@ export const useArticleLinksNavigation = (
     }
   }, dependencies)
 
+  useEffect(() => {
+    const hasGallery = popupState.find(p => p.id === 'Gallery')
+    if (!hasGallery) {
+      if (origin !== 'Article' || popupState.length === 0) {
+        setSoftkeyTrigger(!softkeyTrigger)
+      }
+    }
+  }, [popupState, currentLink])
```

Comment:
This is the addition line that I miss before, could you explain what happen here?

Reply:
Originally, the first `useEffect` in `useArticleLinksNavigation` would watch the `dependencies` passed in to recompute visible links and then consequently trigger the `useSoftkey` hook by updating `links` and `currentLink`. This was fine when changing fontsize direcly with numberic keys (4, 5, 6) but now results in a softkey bug.

After trying a couple of things, the idea I'm proposing is [to decouple](https://github.com/wikimedia/wikipedia-kaios/pull/300#discussion_r551890900) the recomputing of visible links from the actual softkeys setting, and in order to do that I'm introducing this second `useEffect` that decides when to toggle the `softkeyTrigger` by observing the `popupState`